sdk/typescript/runtime/main.go

```diff
@@ -184,29 +179,57 @@ func (t *TypescriptSdk) CodegenBase(ctx context.Context, modSource *dagger.Modul
 	base = base.
 		// Add template directory
 		WithMountedDirectory("/opt/module", dag.CurrentModule().Source().Directory(".")).
-		// Mount users' module with SDK sources and generated client in it.
-		WithMountedDirectory(ModSourceDirPath,
+		// Mount user's module configuration (without sources) and the generated client in it.
+		WithDirectory(ModSourceDirPath,
 			dag.Directory().WithDirectory("/", modSource.ContextDirectory(), dagger.DirectoryWithDirectoryOpts{
 				Include: []string{
-					fmt.Sprintf("%s/**", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/package.json", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/*lock*", t.moduleConfig.subPath),
```

Comment:
Isn't this too broad? e.g. this will upload things like `clock.ts`, `block.js`, `Flock`.

Reply:
True, I can make it more precise

Reply:
Done in [52fd637](https://github.com/dagger/dagger/pull/8236/commits/52fd6376526ba1e9fdec6e1d0c307a48cf370754)